### ISSUE TITLE
VOCDetection test set

### ISSUE THREAD

**Issue by vedaldi** [pytorch/vision#1336]
Hi, `VOCDetection` does not appear to allow loading the test set. It was used to be possible in v0.3.0, but now the option is explicitly rejected. Patching the option check to accept "test" as a possibility still works.

I realise that the test set should be used with care, but it is standard practice in the literature to use the VOC 2007 test set. Test sets are also needed to submit to the evaluation server (albeit for more recent VOCs there are no  test labels available). In practice, not supporting the test set makes it difficult to ship code that reproduces old experiments, as well as to add new comparisons using these older protocols.

In summary, I thin the `VOCDetection` class should accept the test set as an option, providing meaningful labels for the specific case of VOC 2007, and "empty" labels for the other versions.

**Comment by pmeier** [comment]
Related to #1238.

**Comment by fmassa** [comment]
I agree with @vedaldi , and we should fix it.
There was an attempt to fix it in #1238 as pointed out by @pmeier , but it hasn't been merged yet.

I'd be happy to merge a PR fixing this behavior
